app/models/task.py

```diff
@@ -1,5 +1,46 @@
 from app import db
-
+from datetime import datetime
 
 class Task(db.Model):
     task_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String, nullable=False)
+    description = db.Column(db.String, nullable=False)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        if self.goal_id:
+            return {
+                "id": self.task_id,
+                "goal_id": self.goal_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": self.check_if_complete()
+            }
+        else:
+            return {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": self.check_if_complete()
+            }
+
+    @classmethod
+    def from_dict(cls, task_data):
+        return cls(
+            title=task_data["title"],
+            description=task_data["description"],
+            completed_at=None)
+
+    def check_if_complete(self):
```

Comment:
I like how you created this helper to abstract this logic and make it more reusable! There's an even simpler way to accomplish this in one line, by casting the datetime to a boolean:
```python
def check_if_complete(self):
    return bool(self.completed_at)
```